### PR TITLE
Wait for the Cilum CRDs to exist before enabling the load-balancer.

### PR DESCRIPTION
**Pull request by bschimke95**

### CODE REVIEW

File: src/k8s/pkg/component/component.go [hunk 1]
```diff
@@ -67,6 +79,47 @@ func (h *helmClient) Enable(componentName string, values map[string]any) error {
...
+			return err
+		}
+
+		// TODO: This should not be necessary but it seems that the implementation of helms "Wait" command does
+		// not properly wait for the Pods to come up.
+		for _, resource := range resources {
+			cmd := []string{"k8s", "kubectl", "wait", "--for=condition=ready", "pod", "--all", "-n", resource.Namespace, "--timeout", "30m"}
+			// Ignore any error that pops up.
+			// An expected error is e.g. "resource not found" for namespaces that do not have pods.
+			utils.RunCommand(ctx, cmd...)
```
Comment (neoaggelos) [inline src/k8s/pkg/component/component.go:119]: this should not be needed, and the `utils.RunCommand()` is hard to write unit tests with. What happens if we remove it?

File: tests/e2e/tests/test_loadbalancer.py [hunk 2]
```diff
@@ -75,9 +73,11 @@ def test_loadbalancer(instances: List[harness.Instance]):
         parent_cidr=instance_default_cidr,
         excluded_ips=[instance_default_ip, tester_instance_default_ip],
     )
-
+    instance.exec(["k8s", "kubectl", "get", "pods", "-A"])
     instance.exec(["k8s", "set", f"load-balancer.cidrs={lb_cidr}"])
-    instance.exec(["k8s", "enable", "load-balancer"])
+    instance.exec(["journalctl", "-u", "snap.k8s.k8sd", "--no-pager"])
+    instance.exec(["k8s", "kubectl", "get", "pods", "-A"])
+    instance.exec(["k8s", "enable", "load-balancer", "--timeout", "30m"])
```
Comment (neoaggelos) [inline tests/e2e/tests/test_loadbalancer.py:80]: is this needed or is it debugging?

File: src/k8s/pkg/component/component.go [hunk 3]
```diff
@@ -49,14 +55,20 @@ func NewHelmClient(snap snap.Snap, initializer HelmConfigProvider) (*helmClient,
 		initializer = &defaultHelmConfigProvider{restClientGetter: snap.KubernetesRESTClientGetter}
 	}
 
+	k8sClient, err := k8s.NewClient(snap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create k8s client: %w", err)
+	}
+
 	return &helmClient{
+		k8sClient:   k8sClient,
```
Comment (neoaggelos) [inline src/k8s/pkg/component/component.go:64]: We don't use this client, do we? Let's revert this

**Review by neoaggelos — COMMENTED** [review]
Left some comments

File: src/k8s/pkg/component/loadbalancer.go [hunk 4]
```diff
@@ -39,6 +39,25 @@ func UpdateLoadBalancerComponent(ctx context.Context, s snap.Snap, isRefresh boo
...
+		if err != nil {
+			return false, fmt.Errorf("failed to check CRDs: %w", err)
+		}
+
+		for _, sg := range serverGroups.Groups {
+			if sg.Name == "cilium.io" {
+				return true, nil
+			}
+		}
+		return false, nil
```
Comment (neoaggelos) [inline src/k8s/pkg/component/loadbalancer.go:58]: let's specifically check for the crds we care about. use `clientset.Discovery().ServerResourcesForGroupVersion()` and check for the l2 announcement or bgp peering policy crds

**Review by neoaggelos — COMMENTED** [review]
It would be best to split the interactive bootstrap or the context timeout fixes in separate PRs

File: src/k8s/pkg/component/loadbalancer.go (hunk 4 above)
Comment (bschimke95) [reply src/k8s/pkg/component/loadbalancer.go]: thanks - forgot to push the changes yesterday.

**Review by neoaggelos — APPROVED** [review]

**Pull request merged**